Assets/VRTK/Source/Scripts/Interactions/VRTK_InteractiveHaptics.cs

```diff
@@ -0,0 +1,175 @@
+            }
+        }
+
+        protected virtual void OnDisable()
+        {
+            if(objectToAffect != null)
+            {
+                objectToAffect.ValueChanged -= Interact;
+            }
+
+            if (interactableObject != null)
+            {
+                VRTK_InteractableObject.InteractionType start, stop;
+
+                if (interactionType == InteractionType.WhileGrabbing)
+                {
+                    stop = VRTK_InteractableObject.InteractionType.Ungrab;
+                    start = VRTK_InteractableObject.InteractionType.Grab;
+                }
+                else
+                {
+                    stop = VRTK_InteractableObject.InteractionType.Untouch;
+                    start = VRTK_InteractableObject.InteractionType.Touch;
+                }
+
+                interactableObject.UnsubscribeFromInteractionEvent(stop, StopHaptics);
+                interactableObject.UnsubscribeFromInteractionEvent(start, StartHaptics);
+            }
+        }
+
+        private void Start()
+        {
+            if(interactableObject == null)
+            {
+                interactableObject = GetComponentInParent<VRTK_InteractableObject>();
+            }
+            
+            if (interactableObject != null)
+            {
+                VRTK_InteractableObject.InteractionType start, stop;
+
+                if (interactionType == InteractionType.WhileGrabbing)
+                {
+                    stop = VRTK_InteractableObject.InteractionType.Ungrab;
+                    start = VRTK_InteractableObject.InteractionType.Grab;
+                }
+                else
+                {
+                    stop = VRTK_InteractableObject.InteractionType.Untouch;
+                    start = VRTK_InteractableObject.InteractionType.Touch;
+                }
+
+                interactableObject.SubscribeToInteractionEvent(stop, StopHaptics);
+                interactableObject.SubscribeToInteractionEvent(start, StartHaptics);
+            }
+            else
+            {
+                VRTK_Logger.Error(VRTK_Logger.GetCommonMessage(VRTK_Logger.CommonMessageKeys.REQUIRED_COMPONENT_MISSING_FROM_GAMEOBJECT, "VRTK_InteractiveHaptics", "VRTK_InteractableObject", "the parent"));
+            }
+        }
+
+        protected virtual void StopHaptics(object sender, InteractableObjectEventArgs e)
+        {
+            controller = null;
+        }
+
+        protected virtual void StartHaptics(object sender, InteractableObjectEventArgs e)
+        {
+            controller = VRTK_ControllerReference.GetControllerReference(e.interactingObject);
+        }
+
+        private void Interact(object sender, ControllableEventArgs e)
+        {
+            VRTK_ControllerReference controllerReference = null;
```

Comment:
@thestonefox @bddckr maybe you guys could help me out here?

I need to figure out which controller is interacting with the `VRTK_BaseControllable` object at `Interact` time in order to send the pulse. When using rotators/sliders with grab, I seem to only be able to get the controller reference via the `VRTK_InteractableObject` as `interactingTouchScript` is `null`. When using the Pusher classes, `interactingTouchScript` is set, and I can use that to get the controller reference. 

If this was always set to the interacting controller alias, even on grab, then I could get rid of this conditional check as well as all of this:

https://github.com/thestonefox/VRTK/pull/1734/files#diff-5a5fc1a08788396924878f51d1472f61R101

Am i missing an easier solution here?


If